sklearn/utils/parallel.py

```diff
@@ -134,8 +147,11 @@ def __call__(self, *args, **kwargs):
                 ),
                 UserWarning,
             )
-            config = {}
-        with config_context(**config):
+
+        with (
+            config_context(**config),
+            _warning_filter_context(warning_filters=warning_filters),
+        ):
```

Comment:
Rather than creating our own context manager, maybe it would be slightly simpler to use `warnings.catch_warnings` and set `warning.filters = warning_filters`, i.e. something like this:

```py
with config_context(**config), warnings.catch_warnings():
    warnings.filters = warnings_filters
    return self.function(*args, **kwargs)
```